Core/src/org/sleuthkit/autopsy/geolocation/datamodel/Waypoint.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Autopsy Forensic Browser
+ *
+ * Copyright 2019 Basis Technology Corp.
+ * contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.geolocation.datamodel;
+
+import java.util.Map;
+import org.openide.util.NbBundle.Messages;
+import org.sleuthkit.datamodel.AbstractFile;
+
+/**
+ * The basic details of a waypoint.
+ *
+ */
+public interface Waypoint {
+
+    /**
+     * Get the timestamp for this BlackboardArtifact.
+     *
+     * @return Timestamp in epoch seconds or null if none was set.
+     */
+    Long getTimestamp();
+
+    /**
+     * Get the label for this point object.
+     *
+     * @return String label for the point or null if none was set
+     */
+    String getLabel();
+
+    /**
+     * Get the latitude for this point.
+     *
+     * @return Returns the latitude for the point or null if none was set
```

Comment:
You know, I have struggled with this.  Yes, I think a way point should have at a minimum of a longitude and latitude.  None of the other parameters matter.  

This that said, what should be done if an artifact does not have both parameters?  The old KmlReport code would set any missing latitude or longitude to 0.  Since 0,0 is a valid longitude\latitude that does not seem like the correct thing to do.  

Also of note, not all artifacts that we expect a longitude\latitude on (ie TSK_METADATA_EXIF) have the attributes on them.  

Since we don't get the longitude and latitude until the constructor I have been allowing the objects to be created and then not adding them to the list in the getWaypoint function.

This doesn't feel like the correct this to do, but since its valid that the attribute doesn't have the longitude and latitude throwing an exception if they are missing doesn't seem correct either.